.rubocop.yml

```diff
@@ -350,3 +350,8 @@ Naming/FileName:
 # Enforces boolean parameters with default to be keyword arguments.
 Style/OptionalBooleanParameter:
   Enabled: false
+
+Lint/UnusedBlockArgument:
+  # Required keyword arguments normally cannot be omitted or renamed:
+  # Example: in `proc{|unused:|}`, `unused` cannot be renamed to `_unused`.
+  AllowUnusedKeywordArguments: true
```

Comment:
I'm investigating if this is something to bring up with the Rubocop gods tomorrow.

It seems like it's intentionally made to ask you to rename/remove unused keyword arguments, with an option to not perform that check, but I suspect that the default behaviour, `AllowUnusedKeywordArguments: false`, is not valid in general.

Reply:
Regardless, we need this specific configuration to move on with our lives.

Reply:
TBH, I'm not surprised. While I like the concept of what rubocop does, I continue to be baffled by its default configuration. At some point I tried to start a conversation around it in https://github.com/rubocop/rubocop/issues/4895 , but the authors prefer to have lots of options enabled by default, and then configure a bunch of exceptions when they don't make sense/break, so the conversation died down.

Reply:
I would like to simplify the rubocop configuration on ddtrace a lot. Right now, I lose a lot of time with nitpicks, especially because the nitpicks trigger but the auto-correct can't fix them.